Revert the change in uintp.adb that changes the checksum of the source
file in the .ali file, causing bug #601133.

Index: b/src/gcc/ada/uintp.adb
===================================================================
--- a/src/gcc/ada/uintp.adb
+++ b/src/gcc/ada/uintp.adb
@@ -239,7 +239,7 @@
 
    function Hash_Num (F : Int) return Hnum is
    begin
-      return Types."mod" (F, Hnum'Range_Length);
+      return Standard."mod" (F, Hnum'Range_Length);
    end Hash_Num;
 
    ---------------
Index: b/src/gcc/ada/gcc-interface/Make-lang.in
===================================================================
--- a/src/gcc/ada/gcc-interface/Make-lang.in
+++ b/src/gcc/ada/gcc-interface/Make-lang.in
@@ -995,8 +995,8 @@
 	$(ECHO) "with Osint; use Osint;" >>tmp-sdefault.adb
 	$(ECHO) "package body Sdefault is" >>tmp-sdefault.adb
 	$(ECHO) "   S0 : constant String := \"$(prefix)/\";" >>tmp-sdefault.adb
-	$(ECHO) "   S1 : constant String := \"$(ADA_INCLUDE_DIR)/\";" >>tmp-sdefault.adb
-	$(ECHO) "   S2 : constant String := \"$(ADA_RTL_OBJ_DIR)/\";" >>tmp-sdefault.adb
+	$(ECHO) "   S1 : constant String := \"$(libdir)/gcc/$(target_noncanonical)/4.4/adainclude/\";" >>tmp-sdefault.adb
+	$(ECHO) "   S2 : constant String := \"$(libdir)/gcc/$(target_noncanonical)/4.4/adalib/\";" >>tmp-sdefault.adb
 	$(ECHO) "   S3 : constant String := \"$(target)/\";" >>tmp-sdefault.adb
 	$(ECHO) "   S4 : constant String := \"/usr/share/ada/adainclude/\";" >>tmp-sdefault.adb
 	$(ECHO) "   function Include_Dir_Default_Name return String_Ptr is" >>tmp-sdefault.adb
